biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java

```diff
@@ -2011,6 +2012,7 @@ private void fixupOldStyleVersions(Attrs attrs) {
 	void augmentImports(Packages imports, Packages exports) throws Exception {
 		List<PackageRef> noimports = Create.list();
 		Set<PackageRef> provided = findProvidedPackages();
+		EE ee = EE.parse(getHighestEE().getEE());
```

Comment:
I think that instead of simply using `getHighestEE()`, it should first look in `requirements.containsKey(ExecutionEnvironmentNamespace.EXECUTION_ENVIRONMENT_NAMESPACE)` to see if an EE requirement has been set manually - if yes, then we should check against the minimum EE specified in that requirement; if no, *then* use `getHighestEE()`.

Reply:
Hmm I am having some trouble with this. 
I have the feeling you are refferering to the `requirements` somewhere around here

https://github.com/bndtools/bnd/blob/3b0b28f2468869ee385ae355d8495f8ee9ae2932/biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java#L1138

and here

https://github.com/bndtools/bnd/blob/3b0b28f2468869ee385ae355d8495f8ee9ae2932/biz.aQute.bndlib/src/aQute/bnd/osgi/Analyzer.java#L1159

but as far as I see it this is executed after `augmentImports(imports, exports)`

So I have the feeling I am misunderstanding. 
Any other pointers? @kriegfrj

Reply:
@kriegfrj I think in the meantime I came up with something. Let me know if this is what you had in mind.
see https://github.com/bndtools/bnd/pull/6508/commits/2cd573d423e3fff52238f9bcbd06bfc033cc629e